src/library_pthread.js

```diff
@@ -682,9 +691,9 @@ var LibraryPThread = {
     // Ask the worker to start executing its pthread entry point function.
 #if ENVIRONMENT_MAY_BE_NODE
     if (ENVIRONMENT_IS_NODE) {
-      // Mark worker as strongly referenced once we start executing a pthread,
-      // so that Node.js doesn't exit while the pthread is running.
-      worker.ref();
+      // Once we start executing a pthread, we need to mark the worker as weakly
+      // referenced, this ensures that Node.js may exit while a pthread is running.
+      worker.unref();
```

Comment:
> Shouldn't we just set them to unreferenced on first creation then?

I considered making the worker weakly referenced during the `'loaded'` event, i.e.:
<details>
  <summary>Patch</summary>
  
```diff
--- a/src/library_pthread.js
+++ b/src/library_pthread.js
@@ -287,11 +287,9 @@ var LibraryPThread = {
         } else if (cmd === 'loaded') {
           worker.loaded = true;
 #if ENVIRONMENT_MAY_BE_NODE
-          // Check that this worker doesn't have an associated pthread.
-          if (ENVIRONMENT_IS_NODE && !worker.pthread_ptr) {
-            // Once worker is loaded & idle, mark it as weakly referenced,
-            // so that mere existence of a Worker in the pool does not prevent
-            // Node.js from exiting the app.
+          if (ENVIRONMENT_IS_NODE) {
+            // Once the worker is loaded, mark it as weakly referenced, so
+            // that its existence does not prevent Node.js from exiting.
             worker.unref();
           }
 #endif
@@ -686,13 +684,6 @@ var LibraryPThread = {
     // in this file, and not from the external worker.js.
     msg.moduleCanvasId = threadParams.moduleCanvasId;
     msg.offscreenCanvases = threadParams.offscreenCanvases;
-#endif
-#if ENVIRONMENT_MAY_BE_NODE
-    if (ENVIRONMENT_IS_NODE) {
-      // Mark worker as weakly referenced once we start executing a pthread,
-      // so that its existence does not prevent Node.js from exiting.
-      worker.unref();
-    }
 #endif
     // Ask the worker to start executing its pthread entry point function.
     worker.postMessage(msg, threadParams.transferList);
````
</details>

But this `'loaded'` event could occur _after_ `pthread_create()` returns, which would mean it could break such constructions:
```c
pthread_create(&thread, NULL, worker_thread, NULL);
_emscripten_thread_set_strongref(thread, 1);
```
(i.e. the same construction that `PROXY_TO_PTHREAD` currently uses).

>  Or are you saying that an unused worker is strongly referenced and used one only weakly?

I'm not sure if I understand this question, but idle/unused workers (i.e. workers not hosting a pthread) are already weakly referenced within the pool.